ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -575,6 +579,113 @@ private JoinNode buildJoin(final Join root, final String prefix, final boolean i
     );
   }
 
+  private void verifyJoin(final Join root, final PlanNode left, final PlanNode right) {
+    if (left.getNodeOutputType() == DataSourceType.KSTREAM) {
+      if (right.getNodeOutputType() == DataSourceType.KTABLE) {
+
+        // stream-table join detected
+
+        if (root.getInfo().getType().equals(JoinType.OUTER)) {
+          throw new KsqlException("Invalid join type:"
+              + " full-outer join not supported for stream-table joins.");
+        }
+
+        if (joinOnNonKeyAttribute(root.getInfo().getRightJoinExpression(), right)) {
+          throw new KsqlException("Invalid join condition:"
+              + " stream-table joins require to join on the table's primary key.");
+        }
```

Comment:
If we have `ON stream.attributes = table.someAttribute` this conditions verifies `table.someAttribute` and would throw, because we can only use `table.key` for stream-table join. It does not matter here, if we use the stream-side key or some other attribute.

Reply:
got it! (same comment as below then, let's call out what the key is and what we tried to join on)